File: java/word-count/src/main/java/WordCount.java
import java.util.HashMap;

public class WordCount {
    Integer count = 0;

    public HashMap<String, Integer> phrase(String words) {
        HashMap<String, Integer> counts = new HashMap<String, Integer>();
        for (String word : words.split(" ")) {
            String preppedWord = word.toLowerCase().replaceAll("\\W", "");
            if (preppedWord.length() > 0) {
                counts.put(preppedWord, counts.getOrDefault(preppedWord, 0) + 1);
            }
        }
        return counts;
    }
}

File: ruby/word-count/word_count.rb
class Phrase
  attr_reader :word_count
  def initialize(text)
    @word_count = count(text)
  end

  def count(text)
    text.gsub(/( ')|(' )/, ' ')
      .downcase
      .split(/[^\w']+/)
      .reject(&:empty?).reduce(Hash.new) do |counts, word|
      counts.merge(word => counts.fetch(word, 0) + 1)
    end
  end
end
